cwl/src/main/scala/cwl/package.scala

```diff
@@ -81,10 +82,14 @@ package object cwl extends TypeAliases {
 
   type WdlTypeMap = Map[String, WomType]
 
+  type RequirementsValidator = Requirement => ErrorOr[Requirement]
```

Comment:
This is the current shape for a `Requirement` validator function. It takes a `Requirement` and opines on its acceptability, returning the same `Requirement` in a `Valid` or an explanation for why it was unacceptable in an `Invalid[NonEmptyList[String]]`. Unacceptable `Requirement`s need to be handled differently depending on whether they are true `requirements` or merely `hints`. Any unsuccessful validations should be returned to the caller to address the error handling that's beyond the scope of this code.